qiskit/qasm3/ast.py

```diff
@@ -679,17 +679,19 @@ class ForLoopStatement(Statement):
             | "[" Range "]"
     """
 
-    __slots__ = ("indexset", "parameter", "body")
+    __slots__ = ("indexset", "parameter", "body", "type")
 
     def __init__(
         self,
         indexset: Union[Identifier, IndexSet, Range],
         parameter: Identifier,
         body: ProgramBlock,
+        type: ClassicalType,
```

Comment:
it is a good practice to avoid build-in name shadowing.
```suggestion
        type_: ClassicalType,
```